### ISSUE TITLE
FP S3236: Passthrough parameters should be compliant

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#6014]
### Description

S3236 is raised on the arguments to `TraceEvent` below. The rule should not raise if the argument passed is an IParameterSymbol with the same attribute applied:

1. Get the parameter of TraceEvent fitting to the argument and get the applied `[Caller...]` attribute 
2. Check if the argument passed is an IParameterSymbol and if so if IParameterSymobl.Attributes contain the attribute from 1.

### Repro steps

```cs
public static void Critical(this IDiagnosticsSource self,
    string message,
    [CallerMemberName] string member = null,
    [CallerFilePath] string file = null,
    [CallerLineNumber] int line = 0)
    {
        self.TraceEvent(LogLevel.Critical, eventId: 0, message: message, member: member, file: file, line: line);
    }
```

### Expected behavior

S3236 should not be raised on the `file: file, line: line` arguments.

### Actual behavior

S3236 is raised.

### Known workarounds

None.

### Related information

See also https://peach.sonarsource.com/issues?resolved=false&rules=csharpsquid%3AS3236 Most of the violations fall into that bucket.

**Comment by martin-strecker-sonarsource** [comment]
Peach validation: 57 issues (FPs) resolved. 3 FPs remain (`lineNumber ?? 0`). No TPs on peach right now.
